src/framework/asset/asset-localized.js

```diff
@@ -12,10 +12,17 @@ class LocalizedAsset extends EventHandler {
         this._autoLoad = false;
         this._disableLocalization = false;
 
+        /** @type {number} */
+        // @ts-ignore
```

Comment:
This PR does introduce quite a lot of `@ts-ignore`s into the codebase. What's the plan here?

Reply:
A lot of these classes will have to be changed to properly represent their types. I dont like having them either but its whole separate job it itself and I dont want to bundle it together in this PR.